x-pack/legacy/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -4,240 +4,89 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import {
-  EuiButtonEmpty,
-  EuiContextMenuItem,
-  EuiContextMenuPanel,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiIcon,
-  EuiPopover,
-  EuiLink
-} from '@elastic/eui';
-import url from 'url';
+import { EuiButtonEmpty } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import React, { useState, FunctionComponent } from 'react';
-import { pick } from 'lodash';
+import React, { FunctionComponent, useState } from 'react';
+import styled from 'styled-components';
+import {
+  ActionMenu,
+  ActionMenuDivider,
+  Section,
+  SectionLink,
+  SectionLinks,
+  SectionSubtitle,
+  SectionTitle
+} from '../../../../../../../plugins/observability/public';
 import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
-import { DiscoverTransactionLink } from '../Links/DiscoverLinks/DiscoverTransactionLink';
-import { InfraLink } from '../Links/InfraLink';
-import { useUrlParams } from '../../../hooks/useUrlParams';
-import { fromQuery } from '../Links/url_helpers';
 import { useApmPluginContext } from '../../../hooks/useApmPluginContext';
-
-function getInfraMetricsQuery(transaction: Transaction) {
-  const plus5 = new Date(transaction['@timestamp']);
-  const minus5 = new Date(plus5.getTime());
-
-  plus5.setMinutes(plus5.getMinutes() + 5);
-  minus5.setMinutes(minus5.getMinutes() - 5);
-
-  return {
-    from: minus5.getTime(),
-    to: plus5.getTime()
-  };
-}
-
-function ActionMenuButton({ onClick }: { onClick: () => void }) {
-  return (
-    <EuiButtonEmpty iconType="arrowDown" iconSide="right" onClick={onClick}>
-      {i18n.translate('xpack.apm.transactionActionMenu.actionsButtonLabel', {
-        defaultMessage: 'Actions'
-      })}
-    </EuiButtonEmpty>
-  );
-}
+import { useLocation } from '../../../hooks/useLocation';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { px, units } from '../../../style/variables';
+import { getSections } from './sections';
 
 interface Props {
   readonly transaction: Transaction;
 }
 
-interface InfraConfigItem {
-  icon: string;
-  label: string;
-  condition?: boolean;
-  path: string;
-  query: Record<string, any>;
-}
-
-export const TransactionActionMenu: FunctionComponent<Props> = (
-  props: Props
-) => {
-  const { transaction } = props;
-
+const ActionMenuButton = ({ onClick }: { onClick: () => void }) => (
+  <EuiButtonEmpty iconType="arrowDown" iconSide="right" onClick={onClick}>
+    {i18n.translate('xpack.apm.transactionActionMenu.actionsButtonLabel', {
+      defaultMessage: 'Actions'
+    })}
+  </EuiButtonEmpty>
+);
+
+const SectionContainer = styled.div`
+  margin-bottom: ${px(units.plus)};
+  &:last-of-type {
+    margin-bottom: 0;
+  }
+`;
```

Comment:
Is this not something that is also handled by the shared component? How do the other plugins handle it?

Reply:
it should, but it is not. Metrics plugin doesn't have more than one section on their menu.
<img width="195" alt="Screenshot 2020-02-05 at 10 26 21" src="https://user-images.githubusercontent.com/55978943/73828668-f91fcf00-4801-11ea-8e9c-3e29982698ba.png">

Maybe I should do this change in the shared component, WDYT?

Reply:
yeah that would be great!